pkg/controller/configmap/config.go

```diff
@@ -316,6 +297,102 @@ func validateConfig(cfg *ini.File, sec string, key string) error {
 	return nil
 }
 
+// Validate node agent specific options
+func validateNodeAgentOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	var err error
+	// Check MTU value
+	mtu := cfg.Section("nsx_node_agent").Key("mtu").Value()
+	_, err = strconv.Atoi(mtu)
+	if err != nil {
+		appendErrorIfNotNil(&errs, errors.Wrapf(err, "mtu is invalid"))
+	}
+	return errs
+}
+
+// Validate common options for both ncp and node agent
+func validateCommonOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	data := &configmap.Data
+	var err error
+	// Check DEFAULT section log_file option
+	if cfg.Section("DEFAULT").Key("log_file").Value() != "" {
+		log.Info(fmt.Sprintf("DEFAULT section option: log_file is not allowed to change, current value:%s, reverted to default value:None",
+			cfg.Section("DEFAULT").Key("log_file").Value()))
+		// Revert DEFAULT log_file to default value
+		appendErrorIfNotNil(&errs, fillDefault(cfg, "DEFAULT", "log_file", "", true))
+		// Write config back to ConfigMap data
+		(*data)[operatortypes.ConfigMapDataKey], err = iniWriteToString(cfg)
+		appendErrorIfNotNil(&errs, err)
+	}
+	return errs
+}
+
+// Validate MP specific options
+func validateMPOptions(configmap *corev1.ConfigMap, cfg *ini.File) []error {
+	errs := []error{}
+	data := &configmap.Data
+	var err error
+	if cfg.Section("nsx_v3").Key("policy_nsxapi").Value() == "True" {
```

Comment:
Put section and key into a map, so we can also add new options in the future

Reply:
good suggestion, thanks